### ISSUE TITLE
Capybara::RackTest::Driver respect_data_method option not "data-turbo-method" aware

### ISSUE THREAD

**Issue by ybakos** [teamcapybara/capybara#2543]
## Meta
Capybara Version:
3.36.0
Driver Information (and browser if relevant):
CapyBara::RackTest::Driver

## Expected Behavior

Prior to Rails 7, we can configure Capybara's RackTest driver to simulate clicking on hyperlinks that use the Rails `method: :foo` option, which generates an anchor tag with the `data-method` attribute.

```
# To enable delete links (via data-method attribute) w/o javascript.
  Capybara.register_driver :rack_test do |app|
    Capybara::RackTest::Driver.new(app, respect_data_method: true)
  end
```

For example, delete links rendered with `link_to 'Delete', delete_foo_path(@foo), method: :delete`. Within a system test, this enables us to `click_on 'Delete'` successfully.

In Rails 7, the new anchor tag attribute is `data-turbo-method`, and it seems that the Capybara::RackTest::Driver is not sensitive to this attribute when carrying out a `click_on`.

We expect, when `click_on` a link with the `data-turbo-method` attribute, that it "respects the data method" just as it does with the pre-Rails-7 `data-method` attribute.

## Actual Behavior

`click_on` results in a GET request

## Steps to reproduce

Configure the system tests to use the Capybara RackTest driver, and enable the `respect_data_method` option.

```
# To enable delete links (via data-method attribute) w/o javascript.
  Capybara.register_driver :rack_test do |app|
    Capybara::RackTest::Driver.new(app, respect_data_method: true)
  end
```

Create a route and view that results in a hyperlink that simulates sending a PATCH request with the Rails 7 `data-turbo-method` option:

```
    resources :scholarships do
      patch 'publish', on: :member
```

and:

```
= link_to 'publish', publish_scholarship_path(@scholarship), data: { 'turbo-method': :patch }
```

Write a system test that instructs Capybara to `click_on` the link:

```
click_on 'publish'
```

Observer that Capybara is sending a GET instead of a simulated PATCH.

ActionController::RoutingError: No route matches [GET] "/scholarships/123/publish"

**Comment by ybakos** [comment]
https://github.com/teamcapybara/capybara/blob/d05ae5c595bb60237629f710ae33cb63ed4a7790/lib/capybara/rack_test/node.rb#L247

Could we change...

```
  def follow_link
    method = self['data-method'] if driver.options[:respect_data_method]
    method ||= :get
    driver.follow(method, self[:href].to_s)
  end
```

To:

```
def follow_link
    if driver.options[:respect_data_method]
      method = self['data-method'] 
      method ||= self['data-turbo-method']
    end
    method ||= :get
    driver.follow(method, self[:href].to_s)
  end
```

In addition, here are two workarounds I can think of, in case others encounter this issue.

1. Use both the older `method: :foo` option in addition to `data: {'turbo-method': :foo}`.
2. Use `button_to method: :foo` instead of `link_to ... method: ...`

**Comment by twalpole** [comment]
Implemented in cc451e77b2fb6588e1d00814d0361e63bb66b86c
